README.md
# AlphaVantage
[![Known Vulnerabilities](https://snyk.io/test/github/Skitionek/alpha-vantage-data-source/badge.svg?targetFile=package.json)](https://snyk.io/test/github/Skitionek/alpha-vantage-data-source?targetFile=package.json)
[![Intergation test](https://github.com/skitionek/alpha-vantage-data-source/workflows/Intergation%20test/badge.svg)](https://github.com/Skitionek/alpha-vantage-data-source/actions?query=workflow%3A%22Intergation+test%22)
[![CI](https://github.com/skitionek/alpha-vantage-data-source/workflows/CI/badge.svg)](https://github.com/Skitionek/alpha-vantage-data-source/actions?query=workflow%3ACI)

This is a simple wrapper around the [Alpha Vantage API](https://www.alphavantage.co/documentation/) hosted on [NPM](https://www.npmjs.com/package/alphavantage). I have no affiliation with AlphaVantage.

## Contents

1. [Installation](#installation)
2. [Usage](#usage)
    - [Data](#data)
    - [Forex](#forex)
    - [Crypto](#crypto)
    - [Technicals](#technicals)
    - [Performance](#performance)
    - [Util](#util)
3. [Coverage](#coverage)
4. [Testing](#testing)
5. [Mocking](#mocking)
6. [Data structure overview](#data_structure_overview)
7. [Contributing](#contributing)
8. [Credits](#credits)


## Installation
```bash
npm i alpha-vantage-data-source
```

## Usage

The library is initialised in straight forward manner by simply providing AlphaVantage access key:
```javascript
const alpha = require('alpha-vantage-data-source')({ key: 'demo' });
```
This access key might be set by environmental variable 'AV_KEY' if needed.

Once this library ins initialised, data can be accessed by navigating to functions corresponding to AlphaVantage API:
```javascript
alpha.data.intraday(`msft`)
alpha.data.batch([`msft`, `aapl`])
alpha.forex.rate('btc', 'usd')
alpha.crypto.daily('btc', 'usd')
alpha.technical.sma(`msft`, `daily`, 60, `close`)
alpha.performance.sector()
```

Whole list of available functions is as follows"

### Data

See [Alpha Vantage](https://www.alphavantage.co/documentation/#time-series-data) for the parameters.
```javascript
alpha.data.intraday(symbol, outputsize, datatype, interval)
alpha.data.daily(symbol, outputsize, datatype, interval)
alpha.data.daily_adjusted(symbol, outputsize, datatype, interval)
alpha.data.weekly(symbol, outputsize, datatype, interval)
alpha.data.weekly_adjusted(symbol, outputsize, datatype, interval)
alpha.data.monthly(symbol, outputsize, datatype, interval)
alpha.data.monthly_adjusted(symbol, outputsize, datatype, interval)
alpha.data.quote(symbol, outputsize, datatype, interval)
alpha.data.search(keywords)
alpha.data.batch([symbol1, symbol2..])
```

### Forex

See [Alpha Vantage](https://www.alphavantage.co/documentation/#fx) for the parameters.
```javascript
alpha.forex.rate(from_currency, to_currency)
```

### Crypto

See [Alpha Vantage](https://www.alphavantage.co/documentation/#digital-currency) for the parameters.

```javascript
alpha.crypto.daily(symbol, market)
alpha.crypto.weekly(symbol, market)
alpha.crypto.monthly(symbol, market)
```

### Technicals

See [Alpha Vantage](https://www.alphavantage.co/documentation/#technical-indicators) for the parameters.
```javascript
alpha.technical.sma(symbol, interval, time_period, series_type)
alpha.technical.ema(symbol, interval, time_period, series_type)
alpha.technical.wma(symbol, interval, time_period, series_type)
alpha.technical.dema(symbol, interval, time_period, series_type)
alpha.technical.tema(symbol, interval, time_period, series_type)
alpha.technical.trima(symbol, interval, time_period, series_type)
alpha.technical.kama(symbol, interval, time_period, series_type)
alpha.technical.mama(symbol, interval, series_type, fastlimit, slowlimit)
alpha.technical.t3(symbol, interval, time_period, series_type)
alpha.technical.macd(symbol, interval, series_type, fastperiod, slowperiod, signalperiod)
alpha.technical.macdext(symbol, interval, series_type, fastperiod, slowperiod, signalperiod, fastmatype, slowmatype, signalmatype)
alpha.technical.stoch(symbol, interval, fastkperiod, slowkperiod, slowdperiod, slowkmatype, slowdmatype)
alpha.technical.stochf(symbol, interval, fastkperiod, fastdperiod, fastdmatype)
alpha.technical.rsi(symbol, interval, time_period, series_type)
alpha.technical.stochrsi(symbol, interval, time_period, series_type, fastkperiod, slowdperiod, fastdmatype)
alpha.technical.willr(symbol, interval, time_period)
alpha.technical.adx(symbol, interval, time_period)
alpha.technical.adxr(symbol, interval, time_period)
alpha.technical.apo(symbol, interval, series_type, fastperiod, slowperiod, matype)
alpha.technical.ppo(symbol, interval, series_type, fastperiod, slowperiod, matype)
alpha.technical.mom(symbol, interval, time_period, series_type)
alpha.technical.bop(symbol, interval)
alpha.technical.cci(symbol, interval, time_period)
alpha.technical.cmo(symbol, interval, time_period, series_type)
alpha.technical.roc(symbol, interval, time_period, series_type)
alpha.technical.rocr(symbol, interval, time_period, series_type)
alpha.technical.aroon(symbol, interval, time_period)
alpha.technical.aroonosc(symbol, interval, time_period)
alpha.technical.mfi(symbol, interval, time_period)
alpha.technical.trix(symbol, interval, time_period, series_type)
alpha.technical.ultosc(symbol, interval, timeperiod1, timeperiod2, timeperiod3)
alpha.technical.dx(symbol, interval, time_period)
alpha.technical.minus_di(symbol, interval, time_period)
alpha.technical.plus_di(symbol, interval, time_period)
alpha.technical.minus_dm(symbol, interval, time_period)
alpha.technical.plus_dm(symbol, interval, time_period)
alpha.technical.bbands(symbol, interval, time_period, series_type, nbdevup, nbdevdn)
alpha.technical.midpoint(symbol, interval, time_period, series_type)
alpha.technical.midprice(symbol, interval, time_period)
alpha.technical.sar(symbol, interval, acceleration, maximum)
alpha.technical.trange(symbol, interval)
alpha.technical.atr(symbol, interval, time_period)
alpha.technical.natr(symbol, interval, time_period)
alpha.technical.ad(symbol, interval)
alpha.technical.adosc(symbol, interval, fastperiod, slowperiod)
alpha.technical.obv(symbol, interval)
alpha.technical.ht_trendline(symbol, interval, series_type)
alpha.technical.ht_sine(symbol, interval, series_type)
alpha.technical.ht_trendmode(symbol, interval, series_type)
alpha.technical.ht_dcperiod(symbol, interval, series_type)
alpha.technical.ht_dcphase(symbol, interval, series_type)
alpha.technical.ht_dcphasor(symbol, interval, series_type)
```

### Performance

See [Alpha Vantage](https://www.alphavantage.co/documentation/#sector-information) for the parameters.
```javascript
alpha.performance.sector()
```

### Util

Data polishing
  - Rewrite data keys to be consistent across all api calls. This is an optional utility you can use with the result of any api call.

```javascript
const polished = alpha.util.polish(data);
```

## Coverage

As of time of witting this library gives access you all data provided by AlphaVantage. Including comprehensive tests, unit as well as integration (integration tests are validating response for all queries available when using 'demo' key). 

## Testing

This package implements extensive testing, which is used to validate all functionally and also that this package remains up to date with currently exposed version of AlphaVantage API. Automatic cron job runs integration tests every day to keep track if API did not change.

## Mocking

Additionally this project also exposes internally used testing mocks. 
```javascript
const alpha_mocks = require('alpha-vantage-data-source/mocks')
```

## Data structure overview

For ease of integration structure of underlying data is summarised in autogenerated file 'ApiTree.json'. 

## Contributing

All contributions are welcome! This is an open source project under the MIT license, see [LICENSE.md](LICENSE.md) for additional information.

## Credits

Core has been forked from:
  - Author: Zack Urben
  - Twitter: https://twitter.com/zackurben (better)
  - Contact: zackurben@gmail.com
yet greatest part in common is it's structure. Current version support full API coverage, requests cache and non duck-typed keys normalisation.


CHANGELOG.md
## [2.3.1](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.3.0...v2.3.1) (2020-08-24)


### Bug Fixes

* residue nulls ([b626f33](https://github.com/Skitionek/alpha-vantage-data-source/commit/b626f337af6f53b5d2cfe6722cbaf1d8296d2078))

# [2.3.0](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.43...v2.3.0) (2020-08-20)


### Features

* use undefined instead of nulls ([8f61570](https://github.com/Skitionek/alpha-vantage-data-source/commit/8f61570307f968855ac0b5d6b2bd239483ea7d62))

## [2.2.43](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.42...v2.2.43) (2020-08-19)


### Bug Fixes

* bump deps (passing tests) add pycharm files for my ease of maintain ([ed9dbf9](https://github.com/Skitionek/alpha-vantage-data-source/commit/ed9dbf9ae78b15e286fcc5bf49d27d100d08dd84))

## [2.2.42](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.41...v2.2.42) (2020-04-30)


### Bug Fixes

* upgrade @babel/polyfill from 7.4.4 to 7.8.7 ([c2389e1](https://github.com/Skitionek/alpha-vantage-data-source/commit/c2389e1a2cb5c776c1b78586061e3a160440fab9))
* upgrade @babel/polyfill from 7.4.4 to 7.8.7 ([fe8986f](https://github.com/Skitionek/alpha-vantage-data-source/commit/fe8986f9886b6b253136631e11ae457e361ab918))
* upgrade dotenv from 8.0.0 to 8.2.0 ([3f35b3f](https://github.com/Skitionek/alpha-vantage-data-source/commit/3f35b3f759f7709597e69fbc157475963d74ad12))
* upgrade dotenv from 8.0.0 to 8.2.0 ([d005b6f](https://github.com/Skitionek/alpha-vantage-data-source/commit/d005b6fd666d0d3c78cabe22e3ecf6dea187e68f))

## [2.2.41](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.40...v2.2.41) (2020-04-30)


### Bug Fixes

* CI to publish to github npm reppo? ([4cdaa35](https://github.com/Skitionek/alpha-vantage-data-source/commit/4cdaa35152c1843adab83388a8835805fba4df0f))
* resign from type names in snaps ([25df3c4](https://github.com/Skitionek/alpha-vantage-data-source/commit/25df3c4b136fa42e04b9b99e021884121bfff847))

## [2.2.40](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.39...v2.2.40) (2020-04-24)


### Bug Fixes

* package.json ([3fd8449](https://github.com/Skitionek/alpha-vantage-data-source/commit/3fd8449b05be60b312b298d29c9cf91d22c08c4f))

## [2.2.39](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.38...v2.2.39) (2020-04-23)


### Bug Fixes

* deps ([3baf9cb](https://github.com/Skitionek/alpha-vantage-data-source/commit/3baf9cb2c1f9f430fc46fc85b3c19883987846ab))

## [2.2.38](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.37...v2.2.38) (2020-04-20)


### Bug Fixes

* polishing ([e9f70b0](https://github.com/Skitionek/alpha-vantage-data-source/commit/e9f70b03fd43cdcde8e1229232519a5e7f019418))

## [2.2.37](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.36...v2.2.37) (2020-04-20)


### Bug Fixes

* cosmetics ([ac39b71](https://github.com/Skitionek/alpha-vantage-data-source/commit/ac39b7122df1782159efec2e54fa8afdaa6f7d9c))
* cosmetics ([1837e85](https://github.com/Skitionek/alpha-vantage-data-source/commit/1837e85c6b063514cc75750f1093c527acd6c744))
* getting CI to work ([068493a](https://github.com/Skitionek/alpha-vantage-data-source/commit/068493a6ab70fcc6fe4b73920b81bcd2fce7b181))

## [2.2.36](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.35...v2.2.36) (2020-04-20)


### Bug Fixes

* getting CI to work ([fd22dec](https://github.com/Skitionek/alpha-vantage-data-source/commit/fd22dec231147363bcffd42fb5b63779772f6f5a))
* getting CI to work ([7575031](https://github.com/Skitionek/alpha-vantage-data-source/commit/75750317dedfed1065a6d486bbeca0ad0797c83d))

## [2.2.35](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.34...v2.2.35) (2020-04-20)


### Bug Fixes

* getting CI to work ([49fbdd5](https://github.com/Skitionek/alpha-vantage-data-source/commit/49fbdd511e7e669a1e0d591a9e40b41656673985))

## [2.2.34](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.33...v2.2.34) (2020-04-20)


### Bug Fixes

* getting CI to work ([07a1e3a](https://github.com/Skitionek/alpha-vantage-data-source/commit/07a1e3ab6a77ca90ee9f235a38efc80dc0d3d3b4))

## [2.2.33](https://github.com/Skitionek/alpha-vantage-data-source/compare/v2.2.32...v2.2.33) (2020-04-20)


### Bug Fixes

* getting CI to work ([4fd0057](https://github.com/Skitionek/alpha-vantage-data-source/commit/4fd0057d257962b6c8b458bafe0a2c6fbe1ba394))


test/cache.test.js
/* DOCUMENT INFORMATION
	- Author:   Dominik Maszczyk
	- Email:    Skitionek@gmail.com
	- Created:  23/04/2019
*/

/* global performance*/

import AlphaVantageAPI from "../src";
import * as variables from "../src/mocks/demoVariableSets";

jest.setTimeout(30000);

const alphaStructure = AlphaVantageAPI.prototype;

describe.each(['data', 'crypto', 'forex', 'performance', 'technical'])("%s", groupKey => {
	describe.each(Object.keys(alphaStructure[groupKey]))("%s", key => {
		let varSets = variables[groupKey][key];
		varSets = Array.isArray(varSets) ? varSets : [varSets];
		it.each(varSets)("%j", async varSet => {
			const alpha = new AlphaVantageAPI({ key: 'demo' });
			const start = performance.now();
			await alpha[groupKey][key](varSet);
			const response1time = performance.now();
			await alpha[groupKey][key](varSet);
			const response2time = performance.now();
			console.log(response1time - start + 1,'>>',response2time - response1time);
			expect(response1time - start + 1).toBeGreaterThan(response2time - response1time)
		});
	})
});



src/mocks/demoVariableSets.js
/* DOCUMENT INFORMATION
	- Author:   Dominik Maszczyk
	- Email:    Skitionek@gmail.com
	- Created:  24/04/2019
*/

// https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=MSFT&interval=5min&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=MSFT&interval=5min&outputsize=full&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=MSFT&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=MSFT&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=MSFT&outputsize=full&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=MSFT&outputsize=full&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_WEEKLY&symbol=MSFT&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_WEEKLY_ADJUSTED&symbol=MSFT&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_MONTHLY&symbol=MSFT&apikey=demo
// https://www.alphavantage.co/query?function=TIME_SERIES_MONTHLY_ADJUSTED&symbol=MSFT&apikey=demo

// https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=MSFT&apikey=demo
// https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=300135.SZ&apikey=demo

// https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=BA&apikey=demo
// https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=Micro&apikey=demo
// https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=sony&apikey=demo
// https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords=xiaomi&apikey=demo
const intraday = [
	{ symbol: "MSFT", interval: "5min" },
	{ symbol: "MSFT", interval: "5min", outputsize: "full" }
];
const daily = [{ symbol: "MSFT" },
	{ symbol: "MSFT", outputsize: "full" }];
const weekly = { symbol: "MSFT" };
const monthly = { symbol: "MSFT" };
const timeSeries = [
	...daily.map(d => ({ ...d, interval: 'daily' })),
	{ ...weekly, interval: 'weekly' },
	{ ...monthly, interval: 'monthly' }
];
export const data = {
	intraday,
	daily,
	daily_adjusted: daily,
	weekly,
	weekly_adjusted: weekly,
	monthly,
	monthly_adjusted: monthly,
	quote: [{ symbol: "MSFT" },
		{
			symbol: "300135.SHZ"
		}],
	search: [
		{ keywords: "BA" },
		{ keywords: "Micro" },
		{ keywords: "sony" },
		{ keywords: "xiaomi" }
	],

	exchangeTimeSeries: [
		...intraday,
		...timeSeries
	],
	exchangeTimeSeries_adjusted: timeSeries
};

// https://www.alphavantage.co/query?function=FX_INTRADAY&from_symbol=EUR&to_symbol=USD&interval=5min&apikey=demo
// https://www.alphavantage.co/query?function=FX_INTRADAY&from_symbol=EUR&to_symbol=USD&interval=5min&outputsize=full&apikey=demo
// https://www.alphavantage.co/query?function=FX_DAILY&from_symbol=EUR&to_symbol=USD&apikey=demo
// https://www.alphavantage.co/query?function=FX_DAILY&from_symbol=EUR&to_symbol=USD&outputsize=full&apikey=demo
// https://www.alphavantage.co/query?function=FX_WEEKLY&from_symbol=EUR&to_symbol=USD&apikey=demo
// https://www.alphavantage.co/query?function=FX_MONTHLY&from_symbol=EUR&to_symbol=USD&apikey=demo
export const ForeignExchangeSymbols = {
	from_symbol: "EUR",
	to_symbol: "USD"
};
export const forex = {
	exchangeRates: [
		{ from_currency: "BTC", to_currency: "CNY" },
		{ from_currency: "USD", to_currency: "JPY" }
	],

	intraday: [
		{ ...ForeignExchangeSymbols, interval: "5min" },
		{ ...ForeignExchangeSymbols, interval: "5min", outputsize: "full" }
	],
	daily: [
		ForeignExchangeSymbols,
		{ ...ForeignExchangeSymbols, outputsize: "full" }
	],
	weekly: ForeignExchangeSymbols,
	monthly: ForeignExchangeSymbols,

	exchangeTimeSeries: [
		{ ...ForeignExchangeSymbols, interval: "5min" },
		{ ...ForeignExchangeSymbols, interval: "5min", outputsize: "full" },
		{ ...ForeignExchangeSymbols, interval: "daily" },
		{ ...ForeignExchangeSymbols, interval: "daily", outputsize: "full" },
		{ ...ForeignExchangeSymbols, interval: "weekly" },
		{ ...ForeignExchangeSymbols, interval: "monthly" }
	]
};

// https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency=BTC&to_currency=CNY&apikey=demo
// https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency=USD&to_currency=JPY&apikey=demo
// https://www.alphavantage.co/query?function=DIGITAL_CURRENCY_DAILY&symbol=BTC&market=CNY&apikey=demo
// https://www.alphavantage.co/query?function=DIGITAL_CURRENCY_WEEKLY&symbol=BTC&market=CNY&apikey=demo
// https://www.alphavantage.co/query?function=DIGITAL_CURRENCY_MONTHLY&symbol=BTC&market=CNY&apikey=demo
export const CurrencyExchangeSymbols = {
	symbol: "BTC",
	market: "CNY"
};
export const crypto = {
	exchangeRates: forex.exchangeRates,

	daily: CurrencyExchangeSymbols,
	weekly: CurrencyExchangeSymbols,
	monthly: CurrencyExchangeSymbols,

	exchangeTimeSeries: [
		{ ...CurrencyExchangeSymbols, interval: "daily" },
		{ ...CurrencyExchangeSymbols, interval: "daily", outputsize: "full" },
		{ ...CurrencyExchangeSymbols, interval: "weekly" },
		{ ...CurrencyExchangeSymbols, interval: "monthly" }
	]
};

// https://www.alphavantage.co/query?function=SMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=SMA&symbol=USDEUR&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=EMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=EMA&symbol=USDEUR&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=WMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=DEMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// TEMA
// https://www.alphavantage.co/query?function=TEMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// TRIMA
// https://www.alphavantage.co/query?function=TRIMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=KAMA&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=MAMA&symbol=MSFT&interval=daily&series_type=close&fastlimit=0.02&apikey=demo
// https://www.alphavantage.co/query?function=VWAP&symbol=MSFT&interval=15min&apikey=demo
// T3
// https://www.alphavantage.co/query?function=T3&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=MACD&symbol=MSFT&interval=daily&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=MACD&symbol=USDEUR&interval=weekly&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=MACDEXT&symbol=MSFT&interval=daily&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=STOCH&symbol=MSFT&interval=daily&apikey=demo
// https://www.alphavantage.co/query?function=STOCH&symbol=USDEUR&interval=weekly&apikey=demo
// https://www.alphavantage.co/query?function=STOCHF&symbol=MSFT&interval=daily&apikey=demo
// https://www.alphavantage.co/query?function=RSI&symbol=MSFT&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=RSI&symbol=USDEUR&interval=weekly&time_period=10&series_type=open&apikey=demo
// https://www.alphavantage.co/query?function=STOCHRSI&symbol=MSFT&interval=daily&time_period=10&series_type=close&fastkperiod=6&fastdmatype=1&apikey=demo
// https://www.alphavantage.co/query?function=WILLR&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=ADX&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=ADX&symbol=USDEUR&interval=weekly&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=ADXR&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=APO&symbol=MSFT&interval=daily&series_type=close&fastperiod=10&matype=1&apikey=demo
// PPO
// https://www.alphavantage.co/query?function=PPO&symbol=MSFT&interval=daily&series_type=close&fastperiod=10&matype=1&apikey=demo
// https://www.alphavantage.co/query?function=MOM&symbol=MSFT&interval=daily&time_period=10&series_type=close&apikey=demo
// https://www.alphavantage.co/query?function=BOP&symbol=MSFT&interval=daily&apikey=demo
// https://www.alphavantage.co/query?function=CCI&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=CCI&symbol=USDEUR&interval=weekly&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=CMO&symbol=MSFT&interval=weekly&time_period=10&series_type=close&apikey=demo
// https://www.alphavantage.co/query?function=ROC&symbol=MSFT&interval=weekly&time_period=10&series_type=close&apikey=demo
// https://www.alphavantage.co/query?function=ROCR&symbol=MSFT&interval=daily&time_period=10&series_type=close&apikey=demo
// https://www.alphavantage.co/query?function=AROON&symbol=MSFT&interval=daily&time_period=14&apikey=demo
// https://www.alphavantage.co/query?function=AROON&symbol=USDEUR&interval=weekly&time_period=14&apikey=demo
// https://www.alphavantage.co/query?function=AROONOSC&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=MFI&symbol=MSFT&interval=weekly&time_period=10&apikey=demo
// TRIX
// https://www.alphavantage.co/query?function=TRIX&symbol=MSFT&interval=daily&time_period=10&series_type=close&apikey=demo
// https://www.alphavantage.co/query?function=ULTOSC&symbol=MSFT&interval=daily&timeperiod1=8&apikey=demo
// https://www.alphavantage.co/query?function=DX&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=MINUS_DI&symbol=MSFT&interval=weekly&time_period=10&apikey=demo
// PLUS_DI
// https://www.alphavantage.co/query?function=PLUS_DI&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=MINUS_DM&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// PLUS_DM
// https://www.alphavantage.co/query?function=PLUS_DM&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=BBANDS&symbol=MSFT&interval=weekly&time_period=5&series_type=close&nbdevup=3&nbdevdn=3&apikey=demo
// https://www.alphavantage.co/query?function=BBANDS&symbol=USDEUR&interval=weekly&time_period=5&series_type=close&nbdevup=3&nbdevdn=3&apikey=demo
// https://www.alphavantage.co/query?function=MIDPOINT&symbol=MSFT&interval=daily&time_period=10&series_type=close&apikey=demo
// https://www.alphavantage.co/query?function=MIDPRICE&symbol=MSFT&interval=daily&time_period=10&apikey=demo
// https://www.alphavantage.co/query?function=SAR&symbol=MSFT&interval=weekly&acceleration=0.05&maximum=0.25&apikey=demo
// TRANGE
// https://www.alphavantage.co/query?function=TRANGE&symbol=MSFT&interval=daily&apikey=demo
// https://www.alphavantage.co/query?function=ATR&symbol=MSFT&interval=daily&time_period=14&apikey=demo
// https://www.alphavantage.co/query?function=NATR&symbol=MSFT&interval=weekly&time_period=14&apikey=demo
// https://www.alphavantage.co/query?function=AD&symbol=MSFT&interval=daily&apikey=demo
// https://www.alphavantage.co/query?function=ADOSC&symbol=MSFT&interval=daily&fastperiod=5&apikey=demo
// https://www.alphavantage.co/query?function=OBV&symbol=MSFT&interval=weekly&apikey=demo
// HT_TRENDLINE
// https://www.alphavantage.co/query?function=HT_TRENDLINE&symbol=MSFT&interval=daily&series_type=close&apikey=demo
// HT_SINE
// https://www.alphavantage.co/query?function=HT_SINE&symbol=MSFT&interval=daily&series_type=close&apikey=demo
// HT_TRENDMODE
// https://www.alphavantage.co/query?function=HT_TRENDMODE&symbol=MSFT&interval=weekly&series_type=close&apikey=demo
// HT_DCPERIOD
// https://www.alphavantage.co/query?function=HT_DCPERIOD&symbol=MSFT&interval=daily&series_type=close&apikey=demo
// HT_DCPHASE
// https://www.alphavantage.co/query?function=HT_DCPHASE&symbol=MSFT&interval=daily&series_type=close&apikey=demo
// HT_PHASOR
// https://www.alphavantage.co/query?function=HT_PHASOR&symbol=MSFT&interval=weekly&series_type=close&apikey=demo
export const technical = {
	sma: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	ema: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	wma: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	dema: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	tema: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	trima: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	kama: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	mama: { symbol: 'MSFT', interval: 'daily', series_type: 'close', fastlimit: 0.02 },
	vwap: { symbol: 'MSFT', interval: '15min' },
	t3: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	macd: { symbol: 'MSFT', interval: 'daily', series_type: 'open' },
	macdext: { symbol: 'MSFT', interval: 'daily', series_type: 'open' },
	stoch: { symbol: 'MSFT', interval: 'daily' },
	stochf: { symbol: 'MSFT', interval: 'daily' },
	rsi: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'open' },
	stochrsi: {
		symbol: 'MSFT',
		interval: 'daily',
		time_period: 10,
		series_type: 'close',
		fastkperiod: 6,
		fastdmatype: 1
	},
	willr: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	adx: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	adxr: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	apo: { symbol: 'MSFT', interval: 'daily', series_type: 'close', fastperiod: 10, matype: 1 },
	ppo: { symbol: 'MSFT', interval: 'daily', series_type: 'close', fastperiod: 10, matype: 1 },
	mom: { symbol: 'MSFT', interval: 'daily', time_period: 10, series_type: 'close' },
	bop: { symbol: 'MSFT', interval: 'daily' },
	cci: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	cmo: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'close' },
	roc: { symbol: 'MSFT', interval: 'weekly', time_period: 10, series_type: 'close' },
	rocr: { symbol: 'MSFT', interval: 'daily', time_period: 10, series_type: 'close' },
	aroon: { symbol: 'MSFT', interval: 'daily', time_period: 14 },
	aroonosc: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	mfi: { symbol: 'MSFT', interval: 'weekly', time_period: 10 },
	trix: { symbol: 'MSFT', interval: 'daily', time_period: 10, series_type: 'close' },
	ultosc: { symbol: 'MSFT', interval: 'daily', timeperiod: 8 },
	dx: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	minus_di: { symbol: 'MSFT', interval: 'weekly', time_period: 10 },
	plus_di: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	minus_dm: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	plus_dm: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	bbands: { symbol: 'MSFT', interval: 'weekly', time_period: 5, series_type: 'close', nbdevup: 3, nbdevdn: 3 },
	midpoint: { symbol: 'MSFT', interval: 'daily', time_period: 10, series_type: 'close' },
	midprice: { symbol: 'MSFT', interval: 'daily', time_period: 10 },
	sar: { symbol: 'MSFT', interval: 'weekly', acceleration: 0.05, maximum: 0.25 },
	trange: { symbol: 'MSFT', interval: 'daily' },
	atr: { symbol: 'MSFT', interval: 'daily', time_period: 14 },
	natr: { symbol: 'MSFT', interval: 'weekly', time_period: 14 },
	ad: { symbol: 'MSFT', interval: 'daily' },
	adosc: { symbol: 'MSFT', interval: 'daily', fastperiod: 5 },
	obv: { symbol: 'MSFT', interval: 'weekly' },
	ht_trendline: { symbol: 'MSFT', interval: 'daily', series_type: 'close' },
	ht_sine: { symbol: 'MSFT', interval: 'daily', series_type: 'close' },
	ht_trendmode: { symbol: 'MSFT', interval: 'weekly', series_type: 'close' },
	ht_dcperiod: { symbol: 'MSFT', interval: 'daily', series_type: 'close' },
	ht_dcphase: { symbol: 'MSFT', interval: 'daily', series_type: 'close' },
	ht_phasor: { symbol: 'MSFT', interval: 'weekly', series_type: 'close' }
};

export const performance = {
	sector: {}
};